examples/custom-client-cxx/cli.cpp

```diff
@@ -0,0 +1,232 @@
+      break;
+    case aklite::cli::StatusCode::DownloadFailure:
+      std::cout << "FAILURE: Unable to download target";
+      break;
+    case aklite::cli::StatusCode::DownloadFailureVerificationFailed:
+      std::cout << "FAILURE: Target downloaded but verification has failed";
+      break;
+    case aklite::cli::StatusCode::DownloadFailureNoSpace:
+      std::cout << "FAILURE: There is no enough free space to download the target";
+      break;
+    case aklite::cli::StatusCode::InstallAlreadyInstalled:
+      std::cout << "FAILURE: Selected target is already installed";
+      break;
+    case aklite::cli::StatusCode::InstallDowngradeAttempt:
+      // Should not hit this error, since force_downgrade is set to true
+      std::cout << "FAILURE: Attempted to install a previous version";
+      break;
+
+    // Possible return codes for install command
+    case aklite::cli::StatusCode::InstallAppsNeedFinalization:
+      std::cout << "SUCCESS: Execute `custom-sota-client run` command to finalize installation";
+      break;
+    case aklite::cli::StatusCode::InstallNeedsRebootForBootFw:
+      std::cout << "FAILURE: Reboot is required before installing the target";
+      break;
+    case aklite::cli::StatusCode::InstallAppPullFailure:
+      std::cout << "FAILURE: Unable read target data, make sure it was pulled";
+      break;
+
+    // Possible return codes for install and run command
+    case aklite::cli::StatusCode::InstallNeedsReboot:
+      std::cout << "SUCCESS: Reboot to finalize installation";
+      break;
+    case aklite::cli::StatusCode::OkNeedsRebootForBootFw:
+      std::cout << "SUCCESS: Reboot to finalize bootloader installation";
+      break;
+    case aklite::cli::StatusCode::InstallRollbackNeedsReboot:
+      std::cout << "FAILURE: Installation failed, rollback initiated but requires reboot to finalize";
+      break;
+
+    // Possible return codes for run command
+    case aklite::cli::StatusCode::NoPendingInstallation:
+      std::cout << "FAILURE: No pending installation to run";
+      break;
+    case aklite::cli::StatusCode::InstallRollbackOk:
+    case aklite::cli::StatusCode::InstallOfflineRollbackOk:
+      std::cout << "FAILURE: Installation failed, rollback performed";
+      break;
+    case aklite::cli::StatusCode::InstallRollbackFailed:
+      std::cout << "FAILURE: Installation failed and rollback operation was not successful";
+      break;
+    case aklite::cli::StatusCode::UnknownError:
+      std::cout << "FAILURE: Unknown error";
+      break;
+
+    default:
+      std::cout << "FAILURE: Unexpected return code " << static_cast<int>(ret);
+      break;
+  }
+  std::cout << std::endl;
+}
+
+int cmd_check(std::string local_repo_path) {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
```

Comment:
Do you think it is possible and makes sense to extract the `AkliteClient` instantiation into a separate function and invoke the client initialization function from each `cmd_*` functions. (I mean lines 106 - 132).

Reply:
At first, my idea was to avoid calling additional functions, to make each `cmd_` more self contained. But once the code has been changed to use the cli api, the boilerplate code ended up being much bigger than the actual relevant code. So, yes, I believe it is now better to have a initialization function that is shared among `cmd_*`. Already prepared it, will force push in a bit.

Reply:
Changes pushed